code/modules/power/supermatter/supermatter.dm

```diff
@@ -35,54 +47,103 @@
 
 	var/explosion_power = 8
 
 	var/lastwarning = 0				// Time in 1/10th of seconds since the last sent warning
+	var/last_zap = 0				// Time in 1/10th of seconds since the last tesla zap
 	var/power = 0
 
 	var/oxygen = 0					// Moving this up here for easier debugging.
 
 	//Temporary values so that we can optimize this
 	//How much the bullets damage should be multiplied by when it is added to the internal variables
 	var/config_bullet_energy = 2
-	//How much of the power is left after processing is finished?
-//	var/config_power_reduction_per_tick = 0.5
 	//How much hallucination should it produce per unit of power?
 	var/config_hallucination_power = 0.1
 
+	var/debug = 0
+
+	var/disable_adminwarn = FALSE
+
+	var/aw_normal = FALSE
+	var/aw_notify = FALSE
+	var/aw_warning = FALSE
+	var/aw_danger = FALSE
+	var/aw_emerg = FALSE
+	var/aw_delam = FALSE
+
 	var/obj/item/device/radio/radio
 
 	//for logging
 	var/has_been_powered = 0
 	var/has_reached_emergency = 0
 
+/obj/machinery/power/supermatter_shard/crystal
+	name = "supermatter crystal"
+	desc = "A strangely translucent and iridescent crystal."
+	base_icon_state = "darkmatter"
+	icon_state = "darkmatter"
+	anchored = TRUE
+	warning_point = 200
+	emergency_point = 2000
+	explosion_point = 3600
+	gasefficency = 0.25
+	explosion_power = 24
+
+
 /obj/machinery/power/supermatter_shard/New()
 	. = ..()
 	poi_list |= src
+	atmos_machinery += src
 	radio = new(src)
 	radio.listening = 0
 	investigate_log("has been created.", "supermatter")
 
 
+/obj/machinery/power/supermatter_shard/proc/handle_admin_warnings()
+	if(disable_adminwarn)
+		return
+
+	// Generic checks, similar to checks done by supermatter monitor program.
+	aw_normal = status_adminwarn_check(SUPERMATTER_NORMAL, aw_normal, "INFO: Supermatter crystal has been energised.", FALSE)
+	aw_notify = status_adminwarn_check(SUPERMATTER_NOTIFY, aw_notify, "INFO: Supermatter crystal is approaching unsafe operating temperature.", FALSE)
+	aw_warning = status_adminwarn_check(SUPERMATTER_WARNING, aw_warning, "WARN: Supermatter crystal is taking integrity damage!", FALSE)
+	aw_danger = status_adminwarn_check(SUPERMATTER_DANGER, aw_danger, "WARN: Supermatter integrity is below 50%!", TRUE)
+	aw_emerg = status_adminwarn_check(SUPERMATTER_EMERGENCY, aw_emerg, "CRIT: Supermatter integrity is below 25%!", FALSE)
+	aw_delam = status_adminwarn_check(SUPERMATTER_DELAMINATING, aw_delam, "CRIT: Supermatter is delaminating!", TRUE)
+
+/obj/machinery/power/supermatter_shard/proc/status_adminwarn_check(var/min_status, var/current_state, var/message, var/send_to_irc = FALSE)
+	var/status = get_status()
+	if(status >= min_status)
+		if(!current_state)
+			log_and_message_admins(message)
+			if(send_to_irc)
+				send2adminirc(message)
+		return TRUE
+	else
+		return FALSE
+
+
 /obj/machinery/power/supermatter_shard/Destroy()
 	investigate_log("has been destroyed.", "supermatter")
 	QDEL_NULL(radio)
 	poi_list.Remove(src)
+	atmos_machinery -= src
```

Comment:
You should probably add a check to reset the emergency lighting if the crystal being deleted was in an emergency state - otherwise it could get stuck on if the crystal was, say, caught in another powerful explosion that destroyed it.